tutorials/containers/using-gitlab-as-a-container-registry/using-gitlab-as-a-container-registry.md

```diff
@@ -0,0 +1,244 @@
+```
+
+Run kubectl get secret regcred just to check, mine looks like this:
+
+```
+apiVersion: v1
+data:
+  .dockerconfigjson: eyJhdXRocyI6eyJyZWdpc3RyeS5naXRsYWIuY29tIjp7InVzZXJuYW1lIjoiazhzIiwicGFzc3dvcmQiOiJ6enJQNk0zUEJkX0tuZE5INmt1cCIsImF1dGgiOiJhemh6T25wNmNsQTJUVE5RUW1SZlMyNWtUa2cyYTNWdyJ9fX0=
+kind: Secret
+metadata:
+  creationTimestamp: "2019-08-30T13:55:11Z"
+  name: regcred
+  namespace: default
+  resourceVersion: "19552"
+  selfLink: /api/v1/namespaces/default/secrets/regcred
+  uid: be61c809-1572-431f-8a73-43c80e934923
+type: kubernetes.io/dockerconfigjson
+```
+
+Now you can create a deployment with our newly uploaded container image:
+
+```
+kubectl create deployment gitlabrepositories --image=registry.gitlab.com/<YOUR_USERNAME>/gitlabregistries
+```
+
+Now wait for it to come up:
+
+```
+watch kubectl get pods
+```
+
+Initially kubernetes will fail to pull the image, and you should see something like this:
+
+```
+gitlabrepositories-86d4b9bf87-q86rx   0/1     ImagePullBackOff    0          3m39s
+```
+
+This is because there's no secret, you need to edit the deployment:
+
+```
+kubectl edit deployment gitlabrepositories
+```
+
+Under the `containers` spec you need to add imagePullSecrets so that it looks something like this:
+```
+     spec:
+       containers:
+       - image: registry.gitlab.com/variabledeclared/gitlabregistries
+         imagePullPolicy: Always
+         name: gitlabregistries
+         resources: {}
+         terminationMessagePath: /dev/termination-log
+         terminationMessagePolicy: File
+       dnsPolicy: ClusterFirst
+       imagePullSecrets:
+       - name: regcred
+```
+
+Check the pods again, your container should now have started.
+
+
+If your pod has started then lets check the logs for logging!
+
+```
+kubectl logs -f <YOUR_POD_NAME>
+```
+
+Hmm, there seems to be a problem:
+
+![No logging on the pod](./images/no_logging.png)
+
+I forgot something! But that’s fine because you have a (semi) automatic process setup now, let's go into the docker file and add this line just before CMD:
+
+```
+ENV PYTHONUNBUFFERED=0
+```
+
+Rebuild:
+
+```
+docker build -t registry.gitlab.com/<YOUR_USERNAME>/gitlabregistries:v1 .
+```
+
+Push:
+
+```
+docker push registry.gitlab.com/variabledeclared/gitlabregistries:v1
+```
+
+And update our deployment with the new image tag:
+```
+   spec:
+       containers:
+       - image: registry.gitlab.com/variabledeclared/gitlabregistries:v1
+         imagePullPolicy: Always
+         name: gitlabregistries
+         resources: {}
+         terminationMessagePath: /dev/termination-log
+         terminationMessagePolicy: File
+       dnsPolicy: ClusterFirst
+       imagePullSecrets:
+       - name: regcred
+       restartPolicy: Always
+       schedulerName: default-scheduler
+       securityContext: {}
+      terminationGracePeriodSeconds: 30
+```
+OK lets check the logs now:
+
+![Program is logging](./images/logging_screenshot.png)
+
+Voila!
+
+# Conclusion
```

Comment:
Please make it a separate section ("That's all folks!"). Again, you can reference one of the latest tutorials. Also, add some interesting links to our products to let the reader continue their story.